### PR TITLE
Use 8x faster mysqldump fork

### PR DESCRIPTION
**Pull request by staabm**
the [latest druidfi/mysqldump-php release](https://github.com/druidfi/mysqldump-php/releases/tag/1.1.0) - which is a PSR4 refactored, and regular maintained variant of https://github.com/ifsnop/mysqldump-php - [is 8-10x faster](https://github.com/druidfi/mysqldump-php/pull/38#issuecomment-1966256736) then the origin upstream

### CODE REVIEW

File: config/composer-35.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
     "symfony/process": ">=3.4.47 <7.0",
     "symfony/filesystem": ">=3.4.47 <7.0",
     "vlucas/phpdotenv": "^4.3",
-    "ifsnop/mysqldump-php": "^2.12"
+    "druidfi/mysqldump-php": "^1.1"
```
Comment (staabm) [inline config/composer-35.json:40]: this dependency is requiring PHP 7.4+

is it fine to raise the min php version?

Comment (lucatume) [reply config/composer-35.json:40]: Version 3.5 must be compatible with PHP 7.1 to 7.4, for this reason, the dependency cannot be changed.
It's fine to update `master` with the new solution, but 3.5 will have to keep the existing one.

Comment (staabm) [reply config/composer-35.json]: thanks for pushing it over the finishing line

File: src/WordPress/Database/MysqlDatabase.php [hunk 2]
```diff
@@ -330,14 +330,8 @@ public function import(string $dumpFilePath): int
...
         try {
-            $dump = new class($this->dsn, $this->dbUser, $this->dbPassword) extends Mysqldump {
-                public function start($filename = '')
-                {
-                    $this->dumpSettings['add-drop-table'] = true;
-                    $this->dumpSettings['add-drop-database'] = true;
-                    return parent::start($filename);
-                }
-            };
+            $dumpSettings = ['add-drop-table'=> true, 'add-drop-database' => true];
```
Comment (staabm) [inline src/WordPress/Database/MysqlDatabase.php:333]: will the composer-35.json file use the same source code? if so, this line here might need a if-class-exist or similar construct so it works with both Mysqldump implementations?

Comment (lucatume) [reply src/WordPress/Database/MysqlDatabase.php:333]: Yes, v3.5 is transpiled from v4 and will use the same code.
I've chased a missing db version update issue that was failing the tests, when that is solved, I will test v3.5 for issues, and will, likely put in place the conditional code.

**Comment by lucatume** [comment]
Thank you @staabm again for your PR. 
I've fixed the remaining issues and merged it.

**Pull request merged**